scripts/Util.py

```diff
@@ -2395,7 +2394,9 @@ def callback(future):
         while nRetry < 120:
             nRetry += 1
 
-            if self.supportsDiscovery():
+            # If the process controller supports discovery or if we have a direct endpoint to it, try to
+            # ping it. This can take a few tries as the controller app might still be starting.
+            if self.supportsDiscovery() or controllerEndpoints is not None:
                 proxy.ice_pingAsync().add_done_callback(callback)
```

Comment:
Should we just remove this `if`, in what case are we not entering this branch and just waiting?